### DISCUSSION TITLE
Setting step output fails after migrating to GITHUB_OUTPUT

### DISCUSSION THREAD

**Discussion by VMelnalksnis** [community/community#37796] [Actions] [answered]
After migrating from `set-output` to `GITHUB_OUTPUT` setting the output fails on windows from a powershell script; works fine from ubuntu runners.

The second line fails now when running from a powershell script:
```
"artifact-name=$archive_name" >> $env:GITHUB_OUTPUT
"artifact=$publish_dir\$archive_name" >> $env:GITHUB_OUTPUT
```
And here are the logs and the run https://github.com/VMelnalksnis/Gnomeshade/actions/runs/3355907527/jobs/5560645936#step:9:395:
```
##[debug]Set output artifact-name = Gnomeshade.Desktop_win10-x64.zip
##[debug]Set output artifact = source\Gnomeshade.Desktop\bin\Release\net6.0\win10-x64\publish\Gnomeshade.Desktop_win10-x64.zip
Error: Unable to process file command 'output' successfully.
Error: Invalid format '牡楴慦瑣渭浡㵥湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥洮楳਍牡楴慦瑣猽畯捲履湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥扜湩剜汥慥敳屜湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥洮楳਍'
##[debug]System.Exception: Invalid format '牡楴慦瑣渭浡㵥湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥洮楳਍牡楴慦瑣猽畯捲履湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥扜湩剜汥慥敳屜湇浯獥慨敤䐮獥瑫灯䤮獮慴汬牥洮楳਍'
##[debug]   at GitHub.Runner.Worker.EnvFileKeyValuePairs.GetEnumerator()+MoveNext()
##[debug]   at GitHub.Runner.Worker.SetOutputFileCommand.ProcessCommand(IExecutionContext context, String filePath, ContainerInfo container)
##[debug]   at GitHub.Runner.Worker.FileCommandManager.ProcessFiles(IExecutionContext context, ContainerInfo container)
```

Is there something I'm missing on using the outputs on Windows?

**Comment by ossanna16** [comment]
Hi there @VMelnalksnis and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by airtower-luna** [comment, marked as answer]
Do any of your values contain multiple lines? Each `name=value` must be exactly one line. If you need a multi-line output, you need to use the [same format as for multi-line environment variables](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings) instead (still writing to the outputs file, obviously).

**Reply by VMelnalksnis** [reply]
It is not multiline as you can see in the debug logs, the value that failed is a filepath, and the equivalent [bash script](https://github.com/VMelnalksnis/Gnomeshade/blob/master/deployment/publish.sh) works, only the [powershell](https://github.com/VMelnalksnis/Gnomeshade/blob/master/deployment/publish.ps1) fails.
Here are the relevant lines from the script:
```powershell
param (
    [Parameter(Mandatory = $true, Position = 0)]
    [System.String]
    $Project,
    [Parameter(Mandatory = $true, Position = 1)]
    [System.String]
    $Runtime
)

$publish_dir = "source\$Project\bin\Release\net6.0\$Runtime\publish"
$archive_name = "${Project}_$Runtime.zip"

"artifact-name=$archive_name" >> $env:GITHUB_OUTPUT
"artifact=$publish_dir\$archive_name" >> $env:GITHUB_OUTPUT
```

**Reply by VMelnalksnis** [reply]
Using the multiline syntax helped even though the value was in a single line.
